api/chronos/views.py

```diff
@@ -0,0 +1,146 @@
+from framework.auth.oauth_scopes import CoreScopes
+from osf.models import ChronosJournal, ChronosSubmission, Preprint
+from osf.external.tasks import update_submissions_status_async
+
+
+class ChronosJournalList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.ALWAYS_PUBLIC]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = ChronosJournalSerializer
+    model_class = ChronosJournal
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+    view_category = 'chronos'
+    view_name = 'chronos-journals'
+
+    def get_default_queryset(self):
+        return ChronosJournal.objects.all()
+
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+
+class ChronosJournalDetail(JSONAPIBaseView, generics.RetrieveAPIView):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.ALWAYS_PUBLIC]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = ChronosJournalSerializer
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+    view_category = 'chronos'
+    view_name = 'chronos-journal-detail'
+
+    def get_object(self):
+        try:
+            return ChronosJournal.objects.get(journal_id=self.kwargs['journal_id'])
+        except ChronosJournal.DoesNotExist:
+            raise NotFound
+
+
+class ChronosSubmissionList(JSONAPIBaseView, generics.ListCreateAPIView, ListFilterMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        SubmissionOnPreprintPublishedOrAdmin,
+    )
+    required_read_scopes = [CoreScopes.CHRONOS_SUBMISSION_READ]
+    required_write_scopes = [CoreScopes.CHRONOS_SUBMISSION_WRITE]
+
+    serializer_class = ChronosSubmissionSerializer
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON,)
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+    view_category = 'chronos'
+    view_name = 'chronos-submissions'
+
+    def get_serializer_class(self):
+        if self.request.method == 'POST':
+            return ChronosSubmissionCreateSerializer
+        else:
+            return ChronosSubmissionSerializer
+
+    def get_default_queryset(self):
+        user = get_user_auth(self.request).user
+        queryset = ChronosSubmission.objects.filter(
+            Q(preprint__guids___id=self.kwargs['preprint_id']) &
+            (
+                Q(preprint___contributors__id=user.id if user else None) |
+                Q(status__in=[3, 4])
+            ),
+        ).distinct()
```

Comment:
@mfraezz Is `distinct()` here required?

Reply:
I'm guessing distinct *is* being used to avoid duplicates in this query, I just meant that sometimes we used subqueries to get rid of duplicates instead.